master_visu_scripts/vtk_plot.py

```diff
@@ -145,20 +134,40 @@ def plot(file,field='density',bounds=None,contours=False,
         plotter.add_mesh(cntrs,color='black',line_width=5)
     
     plotter.view_xy()
+    
+    # Calculate Camera Position from Bounds
+    bounds_array = np.array(bounds_m)
+    xmag = float(abs(bounds_array[1] - bounds_array[0]))
+    ymag = float(abs(bounds_array[3] - bounds_array[2]))
+    aspect_ratio = ymag/xmag
+    
+    plotter.window_size = (1024,int(1024*aspect_ratio))
+    
+    xmid = xmag/2 + bounds_array[0] # X midpoint
+    ymid = ymag/2 + bounds_array[2] # Y midpoint
+    zoom = xmag*aspect_ratio*1.875 # Zoom level - not sure why 1.875 works
 
-#    plotter.window_size = 1200,660
-    plotter.window_size = 300,300
-
-    if camera is not None:
-        plotter.camera_position = camera
-        plotter.camera_set = True
-    if off_screen==False:
-        cpos = plotter.show(screenshot='output')
-        return(cpos)
-    else:
-        img = plotter.screenshot(output,transparent_background=True,
+    position = (xmid,ymid,zoom)
+    focal_point = (xmid,ymid,0)
+    viewup = (0,1,0)
+    
+    camera = [position,focal_point,viewup]
+    
+    plotter.camera_position = camera
+    plotter.camera_set = True
+    
+    # Create image
+    img = plotter.screenshot(transparent_background=True,
                              return_img=True)
-        return(img)
+    
+    # Plot using imshow
+    if ax is None:
+        ax = plt.gca()
+    
+    ax.imshow(img,aspect='equal',extent=bounds)
+    
+    return(ax)

```

Comment:
Now that instead of an image, `ax` is returned, lines 76 and 77 of `stress_plots.py` no longer function.

Reply:
I modified stress_plots.py in 9ad0bf1 so that there is no explicit return on the vp.plot call and the appropriate ax object is specified as a function parameter. This will need to be tested with the original source file.